contracts/SpokePool.sol

```diff
@@ -667,6 +681,80 @@ abstract contract SpokePool is
         _emitFillRelay(relayExecution, fillAmountPreFees);
     }
 
+    /**
+     * @notice Caller signals to the system that they want a refund on this chain, which they set as the
+     * `repaymentChainId` on the original fillRelay() call on the `destinationChainId`. An observer should be
+     * be able to 1-to-1 match the emitted RefundRequested event with the FilledRelay event on the `destinationChainId`.
+     * @dev This function could be used to artificially inflate the `fillCounter`, allowing the caller to "frontrun"
+     * and cancel pending fills in the mempool. This would in the worst case censor fills at the cost of the caller's
+     * gas costs. We don't view this as a major issue as the fill can be resubmitted and obtain the same incentive,
+     * since incentives are based on validated refunds and would ignore these censoring attempts. This is no
+     * different from calling `fillRelay` and setting msg.sender = recipient.
+     * @dev Caller needs to pass in `fillBlock` that the FilledRelay event was emitted on the `destinationChainId`.
+     * This is to make it hard to request a refund before a fill has been mined and to make lookups of the original
+     * fill as simple as possible.
+     * @param destinationToken This chain's destination token equivalent for original deposit destination token.
```

Comment:
Thought bubble: Naming this "destinationToken" might be confusing because we also have "destinationChainId" as used in deposits and fills, and "destination" means something different in these two contexts:
 - destinationChainId refers to the chain where the fill is/was made.
 - destinationToken refers to the chain where the refund will be paid out, and should be the address of the refund token on the refund chain, not the fill token on the destination chain. 

Since this address should refer to the address of the token to be refunded, wdyt about naming this "refundToken" ?

Reply:
this is a good point. I like `refundToken` or `repaymentToken`

Reply:
https://github.com/across-protocol/contracts-v2/pull/264/commits/7fda67d9188e57a6ce7bcb5bb6e1df49a64e3ac4